features/dd-sdk-android-trace/src/main/kotlin/com/datadog/android/trace/internal/domain/event/OtelDdSpanToSpanEventMapper.kt

```diff
@@ -0,0 +1,136 @@
+/*
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+* This product includes software developed at Datadog (https://www.datadoghq.com/).
+* Copyright 2016-Present Datadog, Inc.
+*/
+
+package com.datadog.android.trace.internal.domain.event
+
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.context.NetworkInfo
+import com.datadog.android.core.internal.utils.toHexString
+import com.datadog.android.log.LogAttributes
+import com.datadog.android.trace.model.SpanEvent
+import com.datadog.trace.core.DDSpan
+
+internal class OtelDdSpanToSpanEventMapper(
+    internal val networkInfoEnabled: Boolean
+) : ContextAwareMapper<DDSpan, SpanEvent> {
+
+    // region Mapper
+
+    override fun map(datadogContext: DatadogContext, model: DDSpan): SpanEvent {
+        val serverOffset = datadogContext.time.serverTimeOffsetNs
+        val metrics = resolveMetrics(model)
+        val metadata = resolveMeta(datadogContext, model)
+        return SpanEvent(
+            // we remove the first part as we specifically set an IdGeneratorStrategy with
+            // 64bits length. Our current endpoint does not accept trace ids longer than 64bits
+            traceId = resolveTraceId(model),
+            spanId = model.spanId.toHexString(),
+            parentId = model.parentId.toHexString(),
+            resource = model.resourceName.toString(),
+            name = model.operationName.toString(),
+            service = model.serviceName,
+            duration = model.durationNano,
+            start = model.startTime + serverOffset,
+            error = model.error.toLong(),
+            meta = metadata,
+            metrics = metrics
+        )
+    }
+
+    private fun resolveTraceId(model: DDSpan): String {
+        val traceIdAsHex = model.traceId.toHexString()
+        val exceededLength = traceIdAsHex.length - RUM_ENDPOINT_REQUIRED_TRACE_ID_LENGTH
+        return if (exceededLength > 0) {
+            // we suppress the warning as we do the check before
+            @Suppress("UnsafeThirdPartyFunctionCall")
+            traceIdAsHex.substring(exceededLength)
+        } else {
+            traceIdAsHex
+        }
+    }
+
+    // endregion
+
+    // region internal
+
+    private fun resolveMetrics(event: DDSpan): SpanEvent.Metrics {
+        val metrics = resolveMetricsFromSpanContext(event).apply {
+            this["_dd.agent_psr"] = 1.0f
+            this["_sampling_priority_v1"] = 1
```

Comment:
why do we have it here, but not in `DdSpanToSpanEventMapper`?

Reply:
because there is being added by us directly in the Java code the way I've seen